splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceUnitTestWithTempDirectory.java

```diff
@@ -0,0 +1,72 @@
+package com.splicemachine.derby.test.framework;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/// this class extends SpliceUnitTest so that each test
+/// can access a temporary output directory that is removed after the test is run
+/// the temporary output directory is created only on demand.
+/// might be also doable like SpliceSchemaWatcher extends TestWatcher
+public class SpliceUnitTestWithTempDirectory extends SpliceUnitTest
```

Comment:
Why not merge the content of this class into SpliceUnitTest. We already have _createBadLogDirectory_ there, f.e.
Developers creating new unit tests won't need to differentiate between SpliceUnitTest and SpliceUnitTestWithTempDirectory: either you use the temporary directory or not. Passive overhead is negligible.

Reply:
if preferred, also possible.

Reply:
done